mlflow/projects/__init__.py

```diff
@@ -321,8 +344,12 @@ def run(
 
     if backend == "databricks":
         mlflow.projects.databricks.before_run_validations(mlflow.get_tracking_uri(), backend_config)
-    elif backend == "local" and run_id is not None:
-        backend_config_dict[MLFLOW_LOCAL_BACKEND_RUN_ID_CONFIG] = run_id
+    else:
+        if backend == "local" and run_id is not None:
+            backend_config_dict[MLFLOW_LOCAL_BACKEND_RUN_ID_CONFIG] = run_id
+        if tracking_uri is not None:
+            if _resolve_tracking_uri(tracking_uri) is not None:
```

Comment:
I don't think we need `_resolve_tracking_uri`. It should be sufficient to check if `tracking_uri` is not None`

Reply:
On line 346, `mlflow.projects.databricks.before_run_validations()` is called on `mlflow.get_tracking_uri()`, rather than the specified `tracking_uri`. Can we address that as well?

Reply:
For your first comment, I will remove `_resolve_tracking_uri` completely. The only reason I added it is because I wanted to give the environment higher precedence than the CLI but after our conversation it's apparent to me that you want the CLI to have more power over the Env Vars. Please correct me if I'm wrong @dbczumar